lib/presentation/screens/home/home.dart

```diff
@@ -12,28 +14,76 @@ class MyHomePage extends StatefulWidget {
-          Text(
-            InstrealLocalizations.of(context)!.whatsUpIn,
-            style: Theme.of(context).textTheme.displayLarge,
-          ),
-          Text(
-            InstrealLocalizations.of(context)!.montreal,
-            style: Theme.of(context).textTheme.displayLarge!.copyWith(
-                  color: Theme.of(context).colorScheme.primary,
-                  fontFamily: GoogleFonts.caveat().fontFamily,
+      body: NestedScrollView(
+        headerSliverBuilder: (context, innerBoxIsScrolled) => [
+          SliverAppBar(
+            expandedHeight: 160,
+            collapsedHeight: 160,
+            flexibleSpace: FlexibleSpaceBar(
+              background: Padding(
+                padding: const EdgeInsets.all(16.0),
+                child: Column(
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  children: [
+                    Text(
+                      InstrealLocalizations.of(context)!.whatsUpIn,
+                      style: Theme.of(context).textTheme.displayLarge,
+                    ),
+                    Text(
+                      InstrealLocalizations.of(context)!.montreal,
+                      style: Theme.of(context).textTheme.displayLarge!.copyWith(
+                            color: Theme.of(context).colorScheme.primary,
+                            fontFamily: GoogleFonts.caveat().fontFamily,
+                          ),
+                    ),
+                  ],
                 ),
+              ),
+            ),
+            floating: false,
+            pinned: false,
           ),
-          const PostCard(),
-          const PostCard(),
-          const PostCard(),
-          const PostCard(),
         ],
+        body: FutureBuilder<List<Posting>>(
+          future: postingsRepo.postings,
+          builder: (context, snapshot) => switch (snapshot) {
+            AsyncSnapshot(connectionState: ConnectionState.none) ||
+            AsyncSnapshot(connectionState: ConnectionState.waiting) ||
+            AsyncSnapshot(connectionState: ConnectionState.active) =>
+              const Center(
+                child: CircularProgressIndicator(),
+              ),
+            AsyncSnapshot(connectionState: ConnectionState.done, :final data) =>
+              ListView.builder(
+                itemBuilder: (context, index) => PostCard(
+                  posting: data[index],
+                ),
+                itemCount: data!.length,
+                padding: const EdgeInsets.all(16.0),
+              ),
+          },
+        ),
+      ),
+      floatingActionButton: FloatingActionButton(
+        onPressed: () async {
+          await postingsRepo.add(
+            const Posting(
+              id: "",
+              title: 'title',
+              author: 'author',
+              imageUrl:
+                  'https://i1.wp.com/www.suitcasescholar.com/wp-content/uploads/2012/08/DSC_2583.jpg',
+            ),
+          );
+          setState(() {});
```

Comment:
Don't we want to add a simple state management instead of a set state?

Reply:
Here, I just wanted to trigger re-rendering since it's more of testing code that generates the mock data. And yes, probably we want to make the Posting list as a state instead of using the FutureBuilder eventually.